### PR TITLE
Delete sessions that are older than 90 days

### PR DESCRIPTION
**Pull request by RyanMacG**
As part of our data retention policy we no longer want to retain all
sessions, this allows us to clear them down using an automated schedule.

### CODE REVIEW

File: session_deletion.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    sys.exit()
+
+logger.info("SUCCESS: Connection to RDS mysql instance succeeded")
+
+def delete_old_sessions(_event, _context):
+    cursor = db.cursor()
+
+    cursor.execute('''
+        DELETE FROM session
+            WHERE start < DATE_SUB(NOW(), INTERVAL 90 DAY)
```
Comment (adrianclay) [inline session_deletion.py]: Just had a conversation with @afoldesi-gds about `NOW()` and replication.  Looks like we're safe though.

https://dev.mysql.com/doc/refman/5.7/en/replication-features-functions.html

File: test_session_deletion.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            ''')
+        try:
+            yield
+        finally:
+            cursor.execute("TRUNCATE TABLE session")
+            cursor.close()
+            db.close()
+
+
+    def db_user_names(self, cursor):
```
Comment (adrianclay) [inline test_session_deletion.py]: We're getting a bit of duplication between the two files now.  Wonder if it's worth creating a class for these two?

File: test_session_deletion.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        )
+
+    @pytest.fixture
+    def setup_sessions(self):
+        db = self.db_conn()
+        cursor = db.cursor()
+        cursor.execute('''
+            INSERT INTO session (start, stop, siteIP, username)
+                values
+                (NOW(), DATE_ADD(NOW(), INTERVAL +1 HOUR), '1.2.3.4', 'user1'),
```
Comment (adrianclay) [inline test_session_deletion.py]: An interesting test case. Why 1 hour in the future?

File: test_session_deletion.py (hunk 2 above)
Comment (RyanMacG) [reply test_session_deletion.py]: I feel like creating a new file to abstract ~6 SLOC is a bit of an over optimisation personally

File: test_session_deletion.py (hunk 3 above)
Comment (adrianclay) [reply test_session_deletion.py]: I'd probably have it at the boundary, if we know everything greater than 90 days gets deleted have it 90 days old.

Comment (RyanMacG) [reply test_session_deletion.py]: Was just trying to be semi-realistic with when a session would start/stop

Comment (RyanMacG) [reply test_session_deletion.py]: Will add another that's on the boundary

File: test_session_deletion.py (hunk 2 above)
Comment (adrianclay) [reply test_session_deletion.py]: Yeah, I can see that.

Comment (adrianclay) [reply test_session_deletion.py]: Actually, these two methods are looking at different tables anyway.

File: test_session_deletion.py (hunk 3 above)
Comment (afoldesi-gds) [reply test_session_deletion.py]: We should not be interested in the value of the stop field at all.

Comment (RyanMacG) [reply test_session_deletion.py]: We're not, it's not used in the query. I just populated it for the sake of it really

**Pull request merged**